nifi-commons/nifi-sensitive-property-provider/src/main/java/org/apache/nifi/properties/AwsSecretsManagerSensitivePropertyProvider.java

```diff
@@ -0,0 +1,157 @@
+
+        this.client = client;
+        this.objectMapper = new ObjectMapper();
+    }
+
+    @Override
+    public boolean isSupported() {
+        return client != null;
+    }
+
+    @Override
+    public String protect(final String unprotectedValue, final ProtectedPropertyContext context)
+            throws SensitivePropertyProtectionException {
+        Objects.requireNonNull(context, "Property context must be provided");
+        Objects.requireNonNull(unprotectedValue, "Property value must be provided");
+
+        if (client == null) {
+            throw new SensitivePropertyProtectionException("AWS Secrets Manager Provider Not Configured");
+        }
+
+        try {
+            writeLock.lock();
+            final String secretName = context.getContextName();
+            final Optional<JsonNode> secretKeyValuesOptional = getSecretKeyValues(context);
+            final ObjectNode secretObject = (ObjectNode) secretKeyValuesOptional.orElse(objectMapper.createObjectNode());
+
+            secretObject.put(context.getPropertyName(), unprotectedValue);
+            final String secretString = objectMapper.writeValueAsString(secretObject);
+
+            if (secretKeyValuesOptional.isPresent()) {
+                client.putSecretValue(builder -> builder.secretId(secretName).secretString(secretString));
+            } else {
+                client.createSecret(builder -> builder.name(secretName).secretString(secretString));
+            }
+            return context.getContextKey();
+        } catch (final SecretsManagerException | JsonProcessingException e) {
+            throw new SensitivePropertyProtectionException(String.format("AWS Secrets Manager Secret Could Not Be Stored for [%s]", context), e);
+        } finally {
+            writeLock.unlock();
+        }
+    }
+
+    @Override
+    public String unprotect(final String protectedValue, final ProtectedPropertyContext context)
+            throws SensitivePropertyProtectionException {
+        Objects.requireNonNull(context, "Property context must be provided");
+
+        if (client == null) {
+            throw new SensitivePropertyProtectionException("AWS Secrets Manager Provider Not Configured");
+        }
+        try {
+            readLock.lock();
+
+            String propertyValue = null;
+            final Optional<JsonNode> secretKeyValuesOptional = getSecretKeyValues(context);
+            if (secretKeyValuesOptional.isPresent()) {
+                final JsonNode secretKeyValues = secretKeyValuesOptional.get();
+                final String propertyName = context.getPropertyName();
+                if (secretKeyValues.has(propertyName)) {
+                    propertyValue = secretKeyValues.get(propertyName).textValue();
+                }
+            }
+            if (propertyValue == null) {
+                throw new SensitivePropertyProtectionException(String.format("Property [%s] Not Found in AWS Secrets Manager Secret", context.getContextKey()));
```

Comment:
What do you think about updating this message to separate the Secret Name and the Property Name?
```suggestion
                throw new SensitivePropertyProtectionException(String.format("AWS Secret Name [%s] Property Name [%s] not found", context.getContextName(), context.getPropertyName()));
```